main.py

```diff
@@ -1,54 +1,197 @@
+parser = argparse.ArgumentParser(description="Escolha um classificador para criar um cômite")
+parser.add_argument('classifier', metavar='c', type=int, help='Escolha um classificador para criar um cômite. Opções: 1 - Naive Bayes, 2 - Tree Decision, 3 - Knn, 4 - Heterogeneous')
+parent_dir = "path_for_results"
+datasets_dir = "../FlexConC/datasets"
+datasets = sorted(os.listdir(datasets_dir))
+init_labelled = [0.03, 0.05, 0.08, 0.1, 0.13, 0.15, 0.18, 0.2, 0.23, 0.25]
+
+args = parser.parse_args()
+
+fold_result_acc_final = []
+fold_result_f1_score_final = []
+
+
+for threshold in thresholds:
+    
+    comite = "Comite_Naive_" if args.classifier == 1 else "Comite_Tree_" if args.classifier == 2 else 'Comite_KNN_' if args.classifier == 3 else "Comite_Heterogeneo_"
+
+    path = os.path.join(parent_dir)
+    
+    folder_check_csv = f'path_for_results'
+    os.makedirs(folder_check_csv, exist_ok=True)
+
+    file_check = f'{comite}.csv'
+    check = os.path.join(folder_check_csv, file_check)
+
+    if not os.path.exists(check):
+        with open(f'{folder_check_csv}/{file_check}', 'a') as f:
+            f.write(
+                f'"ROUNDS", "DATASET","LABELLED-LEVEL","ACC","F1-SCORE"'
+            )
+
+    file_check = f'{comite}F.csv'
+    check = os.path.join(folder_check_csv, file_check)
+
+    if not os.path.exists(check):
+        with open(f'{folder_check_csv}/{file_check}', 'a') as f:
+            f.write(
+                f'"DATASET","LABELLED-LEVEL","ACC-AVERAGE","STANDARD-DEVIATION-ACC","F1-SCORE-AVERAGE","STANDARD-DEVIATION-F1-SCORE"'
+        )
+
+    for cr in crs:
+        for labelled_level in init_labelled:
+            for dataset in datasets:
+                comite = Ensemble(SelfFlexCon, cr=cr, threshold=threshold)
+
+
+                fold_result_acc = []
+                fold_result_f1_score = []
+                df = pd.read_csv('datasets/'+dataset, header=0)
+                seed(214)
+                kfold = StratifiedKFold(n_splits=10)
+                fold = 1
+                flag = 1
+                _instances = df.iloc[:,:-1].values #X
+                _target_unlabelled = df.iloc[:,-1].values #Y
+                # _target_unlabelled_copy = _target_unlabelled.copy()
+
+                # round counter
+                rounds = 0
+                for train, test in kfold.split(_instances, _target_unlabelled):
+                    X_train, X_test = _instances[train], _instances[test]
+                    y_train, y_test = _target_unlabelled[train], _target_unlabelled[test]
+                    labelled_instances = round(len(X_train)*labelled_level)
+
+                    rounds += 1
+
+                    if args.classifier != 1 and args.classifier != 2 and args.classifier != 3 and args.classifier != 4:
+                        print('\nOpção inválida! Escolha corretamente...\nOpções: 1 - Naive Bayes, 2 - Tree Decision, 3 - Knn, 4 - Heterogeneous\nEx: python main.py 1\n')
```

Comment:
Esse if tá todo errado...
a melhor forma seria:


```suggestion
                    if args.classifier <= 4 and args.classifier > 0:
                        print(
                            '\nOpção inválida! Escolha corretamente...\nOpções'
                            ': 1 - Naive Bayes, 2 - Tree Decision, 3 - Knn, 4 '
                            '- Heterogeneous\nEx: python main.py 1\n'
                        )

```

e mesmo assim, este não é o melhor local para esse IF, deveria ser por volta da linha 26 quando vc já sabe o que o usuário informou no parser.

Reply:
feito